File: src/study/kt/study08/Step04_multiParameter.java
package study.kt.study08;

public class Step04_multiParameter {
    public static void main(String[] args) {
        // 1) 배열
        String[] strArr = new String[]{"A", "B", "C"};
        printAll(strArr);


        // 2) comma를 이용하여 여러 파라미터 대입
        printAll("A", "B", "C");
    }

    public static void printAll(String... strings) {
        for(String str : strings){
            System.out.println(str);
        }
    }
}

File: src/study/kt/study08/Step04_multiParameter.kt
package study.kt.study08

fun main() {
    printAll("A", "B", "C")

    val array = arrayOf("E", "F", "G")

    // *array -> spread 연산자
    // 배열안에 있는 것들은 ',' 사용하는 것 처럼 꺼내서 사용한다.
    printAll(*array);
}

// 가변인자 vararg
fun printAll(vararg strings: String) {
    for(str in strings){
        println(str)
    }
}
